### DISCUSSION TITLE
Rtl515 Garmin 1040 Flash and Brake Light help

### DISCUSSION THREAD

**Discussion by franz075** [maca88/SmartBikeLights#65] [Q&A] [answered]
Hello, I am trying to obtain the following using on the same mode;
I Would like it to flash when I have a car approaching and I would like a solid brake light when braking.
Other than these 2 conditions the light should stay off. 
I have tried the following code but only the flashing on approaching car works.

###0,73535488::1#2,2!CAR:1:4:0:0I[300]0!:1:0:0:0D=1##3,2:Varia 515:16711680:16777215!2,:-1,Off:0!1,Lampeggio:7#0::#0:1#123!:123!#0#0#B3843##3#0#0

**Comment by maca88** [comment, marked as answer]
Hi,

You can use this configuration:

###0,73535488::1#3,3!BRAKE:1:4:2:0A[-10!CAR:1:7:0:0I]-1]0!:1:0:0:0D=1##3,2:Varia 515:16711680:16777215!2,:-1,Off:0!1,Lampeggio:7#0::#0:1#123!:123!#0#0#B3843##3#0#0

feel free to change the sensitivity of braking (currently it will be activated when slowing more than 10% per second).

**Comment by edinn** [comment]
Hi, I just bought Varia RTL-515 and configuring it with my Edge 540 (non-solar). For some reason, the code above does not work for me - I am getting Error 4. I looked into the configuration, and everything seems fine - obviously, I changed it to Edge 540 and entered my serial number. Everything else is the same. Any idea why I am getting Error 4? Here's my configuration: 

`###0,73535488:1,1341572307:1#3,3!BRAKE:1:4:2:0A[-10!CAR:1:7:0:0I]-1]0!:1:0:0:0D=1##5:Varia 515!Off:0!Solid:4!Peloton:5!Day Flash:7!Night Flash:6#0::#0:1#0#0#B4061##3#0#0`

Thanks.

**Reply by maca88** [reply]
Hi,
Are you using Smart Bike Lights data field or Bike Lights Control widget? Based on the error I would assume that you are most likely using  Bike Lights Control widget, which does not support the Smart mode. If you want control the light by using filters (e.g. radar filter), you need to install Smart Bike Lights data field

**Reply by edinn** [reply]
I am using Smart Bike Lights, but managed to make it work - I removed the serial number and it now it does not throw any error. Not sure why because I checked the serial number multiple times and it was correct. I am not using any other lights. 
Here's my new code:
`###0,73535488::1#3,3!Brake:1:4:2:0A[-10!Car:1:7:0:0I]-1]0!:1:0:0:0D=1##5:Varia 515!Off:0!Solid:4!Peloton:5!Day Flash:7!Night Flash:6#0::#0:1#0#0#B4061##3#0#0`
Thank you!

**Reply by maca88** [reply]
Each ANT+ light has two numbers, Device Id and Serial number. Serial number is usually the longer one. If you have only one light, then there is no need to set the serial number.
